doc/design/diagrams/iceoryx_components_diagram_v3_0_0.puml

```diff
@@ -27,26 +28,40 @@ skinparam frame {
 
 header Eclipse iceoryx component overview
 
+' Platform Library containing platform abstraction with quality level 1
+package iceoryx_platform  {
+   frame "platform library" {
+      component "Platform abstraction" <<module>>
+   }
+}
+
 ' HOOFS: Helpful Objects Optimised For Safety
-' Library containing modern C++ STL constructs
+' Library containing modern C++ STL constructs with quality level 1+
 package iceoryx_hoofs  {
    frame "hoofs library" {
-      component concurrent <<namespace>>
-      component cxx <<namespace>> {
-         component newtype <<namespace>>
-         component set <<namespace>>
-         component algorithm <<namespace>>
-      }
-      component DesignPattern <<namespace>>
-      component errors <<namespace>>
-      component log <<namespace>> {
-         component ffbb <<namespace>>
-      }
-      component units <<namespace>> {
-         component duration_literals <<namespace>>
-
-      }
-      component posix <<namespace>>
+
+      component "Memory & lifetime management" <<module>>
+      component Containers <<module>>
+      component "Common vocabulary types" <<module>>
+      component Filesystem <<module>>
+      component Functional <<module>>
+      component Utility <<module>>
+      component Metaprogramming <<module>>
+      component "Queues & communication" <<module>>
+      component "Threads & sychronisation" <<module>>
+      component "Generalized design patterns & abstractions" <<module>>
+      component "Error handling & logging" <<module>>
+      component Time <<module>>
```

Comment:
Oh no, I looked at the SVG and wanted to write that this file is not yet updated but it's only the SVG that is not updated. Could you please push a rendered version of this file.

I'm still thinking we should somehow clarify that the modules are spread across multiple folders in the filesystem. This information is currently missing and since we want to create separate object libs for the modules, this will be a crucial part of the design. Not quite sure how to do it. A comment would be sufficient. Maybe we find a good one. It should maybe also be done in the `README`. Maybe as as developer note ... something like `This is a logical grouping. The modules structure is replicated for 'concurrent' and 'posix' implementations` ... just in good :sweat_smile: